File: LeetCode/May challenge/Day_11.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        if newColor == image[sr][sc]: return image
        
        q, prevColor = [(sr, sc)], image[sr][sc]
        
        while q:
            r, c = q.pop(0)
            
            if 0 <= r < len(image) and 0 <= c < len(image[0]) and image[r][c] == prevColor:
                image[r][c] = newColor
                q.append((r+1, c))
                q.append((r-1, c))
                q.append((r, c+1))
                q.append((r, c-1))
        
        return image

File: LeetCode/May challenge/Day 11/Day_11.cpp
class Solution {
public:
    void helper(vector<vector<int>>& image, int i, int j, int newColor,int prev)
    {
        if(i < 0 || j < 0 || i >= image.size() || j >= image[i].size() || image[i][j] != prev)
            return;
        image[i][j]=newColor;
        helper(image, i-1, j, newColor, prev);
        helper(image, i+1, j, newColor, prev);
        helper(image, i, j-1, newColor, prev);
        helper(image, i, j+1, newColor, prev);
    }
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        
        int prev=image[sr][sc];
        if(prev!=newColor)
            helper(image, sr, sc, newColor, prev);
        
        return image;
    }
};
